pkg/server/token_minter.go

```diff
@@ -292,6 +299,48 @@ func (s *TokenMintServer) processRequest(r *http.Request, auditEvent *auditEvent
 	return http.StatusOK, accessToken, nil
 }
 
+// validateRepositories checks the set of requested repositories against the allow list
+// to verity that it is authorizaed. Response contains the list of allowed repositories
+// after wild card match expansion.
+func validateRepositories(allowed, requested []string) ([]string, error) {
+	repositories := []string{}
+	// Loop through all of the requested repositories to verifiy that are in the configured
+	// allow list
+	for _, repo := range requested {
+		matched := false
+		for _, allow := range allowed {
+			if matchesAllowed(allow, repo) {
+				repositories = append(repositories, allow)
+				matched = true
+			}
+		}
+		// If there is no match then respond with an error
+		if !matched {
+			return nil, fmt.Errorf("requested repository %q is not in the allow list", repo)
+		}
+	}
+	// Return the set of repositories that are allowed. If no repositories are requested
+	// this will result in an empty list which will work for organization level updates
+	// such as adding members to teams.
+	return repositories, nil
+}
+
+func matchesAllowed(allow string, repo string) bool {
+	switch {
+	// Matches all repositories in the allow list
+	case repo == "*":
```

Comment:
NIT

I might change repo to request or something specific to context, this confused me to which was the star at first.

Same wiht line 309 maybe?

Reply:
done